modules/plugin/shapefile/src/test/java/org/geotools/data/shapefile/shp/ZMHandlersTest.java

```diff
@@ -155,15 +155,15 @@ public void testReadMLine() throws ShapefileException, IOException {
 
     @Test
     public void testReadZLine() throws ShapefileException, IOException {
-        URL url = TestData.url(ShapefileDataStore.class, "mzvalues/zlines.shp");
+        URL url = TestData.url(ShapefileDataStore.class, "mzvalues/linez.shp");
```

Comment:
I think it needed to be the last shape in the file being read to trigger the exception (which is the difference between overrunning the buffer vs just reading random data from the next shape in the file). The existing tests were reading the Z-shapes without M-values and passing because they only ever read the _first_ shape. @taba90 changed the file to only contain a single shape, which would then trigger the exception. (For GEOT-6683 I took a slightly different approach and used the existing file, and added a new test that read _all_ the shapes in the file)